src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/GaussianMixtureModel.java

```diff
@@ -0,0 +1,635 @@
+        private final double prior_beta;  //shrinkage
+        private final double prior_alpha; //dirichlet parameter
+        private final RealVector prior_m;  //prior on the mean vector. Size = dim
+        private final RealMatrix prior_L;  //prior on the inverse variance matrix. Size = dim x dim
+
+        //Parameters fitted by the model. They corresponds to parameters indexed by subscript k in Murphy chapter 21.6.1.3
+        private double param_N;                //sum of the fractional weigths assigned to this gaussian from all the data points
+        private RealVector param_xbar;         //mean vector
+        private RealVector param_m;
+        private RealMatrix param_L;            //precision matrix
+        private RealMatrix param_S;            //covariance matrix
+        private double param_nu;
+        private double param_beta;
+        private double param_alpha;
+        private final List<Double> pVarInGaussian;
+
+        private double pMixtureLog10;        //log10 of the mixture weight for this gaussian
+
+        //caches
+        private double cachedDenomLog10;
+
+        MultivariateGaussian(int numDimensions, double prior_alpha, double prior_beta, double prior_nu, RealVector prior_m, RealMatrix prior_L) {
+            this.dim = numDimensions;
+            this.prior_alpha = prior_alpha;
+            this.prior_beta = prior_beta;
+            this.prior_nu = prior_nu;
+            this.prior_m = prior_m;
+            this.prior_L = prior_L;
+
+            this.param_xbar = new ArrayRealVector(numDimensions); //empty
+            this.param_S = new Array2DRowRealMatrix(numDimensions,numDimensions);   //empty
+            this.param_L = new Array2DRowRealMatrix(numDimensions,numDimensions);   //empty
+            this.param_alpha = prior_alpha;
+            this.param_beta = prior_beta;
+            this.param_nu = prior_nu;
+            this.param_m = prior_m;
+            this.pVarInGaussian = new ArrayList<>();
+        }
+
+        @Override
+        public String toString() {
+            List<String> sb = new ArrayList<>();
+            sb.add("pMixtureLog10:" + pMixtureLog10);
+            sb.add("param_N:" + param_N);
+            sb.add("mu:" + param_xbar.toString());
+            sb.add("param_nu:" + param_nu);
+            sb.add("param_beta:" + param_beta);
+            sb.add("param_alpha:" + param_alpha);
+            sb.add("cachedDenomLog10:" + cachedDenomLog10);
+            sb.add("pVarInGaussian:" + pVarInGaussian);
+
+            return Strings.join(sb, "\n");
+        }
+
+        void initializeRandomXBar(final RandomGenerator rand) {
+            for (int i = 0; i < dim; i++) {
+                param_xbar.setEntry(i, MU_MIN + MU_SPAN * rand.nextDouble());
+            }
+        }
+
+        void initializeRandomS(final RandomGenerator rand) {
+            //This is equiv to drawing from Wishart.
+            //Note: maybe we want empirical cov from kmeans clusters
```

Comment:
I see no reason not to use the empirical covariance.  The iteration is convex and will reach a global optimum, so there's no reason not to initialize parameters with the best possible empirical guess.